sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_helpers_async.py

```diff
@@ -0,0 +1,29 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from azure.core.pipeline.policies import AzureKeyCredentialPolicy, AsyncBearerTokenCredentialPolicy
+from azure.core.credentials import AzureKeyCredential, AzureSasCredential
+
+from .. import _constants as constants
+from .._signature_credential_policy import EventGridSasCredentialPolicy
+
+def _get_authentication_policy_async(credential):
```

Comment:
nit: I think we could avoid duplicating the `_get_authentication_policy` method for async by adding a credential type argument in the sync helper.

```python
def _get_authentication_policy(credential, bearer_token_policy=BearerTokenCredentialPolicy):
    if hasattr(credential, "get_token"):
        return bearer_token_policy(
            credential,
            constants.DEFAULT_EVENTGRID_SCOPE
        )
```

then in the async, we do
```python
_get_authentication_policy(credential, AsyncBearerTokenCredentialPolicy)
```